lib/phlex.rb

```diff
@@ -43,6 +43,38 @@ def self.eager_load
 			end
 		end
 	end
+
+	def self.html(&block)
+		HTML.call do |component|
+			receiver = block.binding.receiver
+
+			receiver.instance_variables.each do |ivar|
+				value = receiver.instance_variable_get(ivar)
+
+				unless component.instance_variable_defined?(ivar)
+					component.instance_variable_set(ivar, value)
+				end
```

Comment:
We can skip early to avoid accessing the ivar if we're not going to assign it. (Don't merge this suggestion, I've got spaces instead of tabs here 🙈)

```suggestion
        next if component.instance_variable_defined?(ivar)
        
				value = receiver.instance_variable_get(ivar)
				component.instance_variable_set(ivar, value)
```